### PR TITLE
Meteor Fix

### PR DESCRIPTION
**Pull request by ElectroJr**
I introduced a bug in #5689 where I never checked if the projectile shooter UID was actually valid. This prevented shooter-less projectiles from exploding (e.g., meteors). This fixes that.

### CODE REVIEW

File: Content.Server/Explosion/EntitySystems/ExplosionSystem.cs [hunk 1]
```diff
@@ -344,19 +344,19 @@ private void FlashInRange(EntityCoordinates epicenter, float flashRange)
...
+            if (entity == null || !entity.Value.IsValid())
             {
                 _logSystem.Add(LogType.Explosion, LogImpact.High, $"Explosion spawned at {text}");
             }
-            else if (user == null)
+            else if (user == null || !user.Value.IsValid())
             {
                 _logSystem.Add(LogType.Explosion, LogImpact.High,
-                    $"{EntityManager.GetEntity(entity.Value)} exploded at {text}");
+                    $"{EntityManager.ToPrettyString(entity.Value)} exploded at {text}");
```
Comment (mirrorcult) [inline Content.Server/Explosion/EntitySystems/ExplosionSystem.cs:354]: i think the string interpolator might automatically toprettystring if you pass it an entityuid, would have to check with vera/smug

**Comment by Zumorica** [comment]
Conflicts

File: Content.Server/Explosion/EntitySystems/ExplosionSystem.cs (hunk 1 above)
Comment (ElectroJr) [reply Content.Server/Explosion/EntitySystems/ExplosionSystem.cs:354]: Sadly you apparently still need to do it explicitly

File: Content.Server/Explosion/EntitySystems/ExplosionSystem.cs [hunk 2]
```diff
@@ -333,8 +333,8 @@ private void FlashInRange(EntityCoordinates epicenter, float flashRange)
             int heavyImpactRange = 0,
             int lightImpactRange = 0,
             int flashRange = 0,
-            EntityUid? entity = null,
-            EntityUid? user = null)
```
Comment (Zumorica) [inline Content.Server/Explosion/EntitySystems/ExplosionSystem.cs:337]: If these are meant to be optional, you need to use nullable `EntityUid`

File: Content.Server/Explosion/EntitySystems/ExplosionSystem.cs [hunk 3]
```diff
@@ -294,7 +294,7 @@ private void FlashInRange(EntityCoordinates epicenter, float flashRange)
             int heavyImpactRange = 0,
             int lightImpactRange = 0,
             int flashRange = 0,
-            EntityUid? user = null,
+            EntityUid user = default,
```
Comment (Zumorica) [inline Content.Server/Explosion/EntitySystems/ExplosionSystem.cs:297]: same here, if it's optional use nullable `EntityUid`

**Review by Zumorica — CHANGES_REQUESTED** [review]

**Review by Zumorica — APPROVED** [review]

**Pull request merged**